src/devices/fancy-node/fancy-node.cc

```diff
@@ -46,6 +69,28 @@ void setup() {
 }
 
 void loop() {
-  state_machine->Tick();
-  led_manager->RunEffect();
+  battery_median_filter.Run();
+  battery_average_filter.Run();
+  if (battery_average_filter.GetFilteredValue() <
+      BatteryVoltageToRawReading(kBatteryDead)) {
+    // TODO: display a battery-low pattern?
+    led_manager->SetGlobalColor(CRGB::Black);
+    led_manager->SetOnboardLed(CRGB(4, 0, 0));
+  } else {
+    // Display battery status on onboard LED for the first 20 seconds
+    if (millis() < 20 * 1000) {
+      uint16_t battery_val = battery_average_filter.GetFilteredValue() -
+                             BatteryVoltageToRawReading(kBatteryDead);
+      uint16_t battery_range = BatteryVoltageToRawReading(kBatteryFull) -
+                               BatteryVoltageToRawReading(kBatteryDead);
+      // Red is hue 0
+      uint8_t hue = (battery_val * HUE_GREEN) / battery_range;
```

Comment:
Nice animation! I think we should shove this in the LedManager startup animation however lets not worry about that right now. (I don't know when we'd do voltage sampling in that case.

Reply:
Ack